astropy/nddata/tests/test_nddata.py

```diff
@@ -371,6 +371,15 @@ def test_param_unit():
     assert nd3.unit == u.km
 
 
+def test_pickle_nddata_with_uncertainty():
+    ndd = NDData(np.ones(3), uncertainty=StdDevUncertainty(np.ones(5), unit=u.m))
+    dumped = pickle.dumps(ndd)
+    ndd_restored = pickle.loads(dumped)
+    assert type(ndd_restored.uncertainty) is StdDevUncertainty
+    assert ndd_restored.uncertainty.parent_nddata is ndd_restored
+    assert ndd_restored.uncertainty.unit == u.m
```

Comment:
For completeness it would be good to test a `NDUncertainty` object with no parent reference. And maybe also a `NDData` object without uncertainty, unless it is already tested elsewhere ?

Reply:
You're right. That also revealed a problem when the uncertainty is pickled by itself instead of pickled indirectly (because the parent was pickled).